### PR TITLE
SF-1145: Add implementation for rBASIS price sourcing

### PR DESCRIPTION
**Pull request by AuthorOfTheSurf**
Ticket:
- https://step-finance.atlassian.net/browse/SF-1145

### CODE REVIEW

File: tests/basis.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+describe("Basis Market Staking", () => {
+  it("calculates price of rBASIS vs BASIS", async () => {
+    const aggregator = new MarketAggregator({
+      cluster: "mainnet-beta",
+      endpoint: process.env.MAINNET_ENDPOINT!,
+    });
+
+    // eslint-disable-next-line no-debugger
+    debugger;
```
Comment (aaronovz1) [inline tests/basis.ts:16]: Shouldn't need this here?

File: tests/basis.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import "mocha";
+
+import { expect } from "chai";
+
+import { MarketAggregator } from "../src";
+import { R_BASIS_MINT } from "../src/sources";
+
+describe("Basis Market Staking", () => {
+  it("calculates price of rBASIS vs BASIS", async () => {
```
Comment (aaronovz1) [inline tests/basis.ts:9]: This test should act more like xSTEP etc and not use MarketAggregator. It should test the BasisSource directly. MarketAggregator is covered in its own test.

File: tests/basis.ts (hunk 1 above)
Comment (AuthorOfTheSurf) [reply tests/basis.ts]: Must be removed, done

File: tests/basis.ts (hunk 2 above)
Comment (AuthorOfTheSurf) [reply tests/basis.ts]: I've re-implemented tests now in this way. Testing vs. expected return value rather than being so integration focused.

**Review by aaronovz1 — APPROVED** [review]

**Pull request merged**